### DISCUSSION TITLE
How to listen to pipeline events (such as BIND)?

### DISCUSSION THREAD

**Discussion by LaP0573** [rexrainbow/phaser3-rex-notes#319] [Q&A]
I would like to listen to pipeline events on the Blur pipeline. However, when I get the pipeline (for example through `scene.plugins.get("rexKawaseBlurPipeline")`) and try to call the `on()` method on it, I get an undefined method error.

Anything I'm doing wrong?

**Comment by rexrainbow** [comment]
`scene.plugins.get("rexKawaseBlurPipeline")` is the plugin instance, not pipeline instance.
Pipeline instance will be returned after
```js
var pipelineInstance = scene.plugins.get('rexKawaseBlurPipeline').add(camera)
```

**Comment by LaP0573** [comment]
Oh okay makes sense! Do you know if there is a way to get a "broad" event on the renderer to know when a pipeline gets bound or applied to sth (camera or game object) ? Or you have to manually register the event on the pipeline every time one is bound/used?

**Reply by rexrainbow** [reply]
Try ['pipelinebind' event](https://github.com/photonstorm/phaser/blob/master/src/renderer/webgl/WebGLPipeline.js#L1071)

```js
pipelineInstance.on('pipelinebind', function(){
    //...
})
```
